From ae75f642859dcfdcfcaaffc617ebe7a68cf0bed7 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 29 Sep 2021 11:03:41 -0400
Subject: [PATCH] daemon: Infer session type from desktop file if user has no
 saved session type

The accountsservice user cache file can specify a session type
associated with the saved session.  This is optional though. If one
isn't specified GDM needs to figure out the session type based on the
list of preferred session types for the system and the session file
itself.

It was failing to do the latter, though.  This commit fixes that.
---
 daemon/gdm-session.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/daemon/gdm-session.c b/daemon/gdm-session.c
index c1e2be008..4b709731d 100644
--- a/daemon/gdm-session.c
+++ b/daemon/gdm-session.c
@@ -1036,6 +1036,8 @@ worker_on_saved_session_name_read (GdmDBusWorker          *worker,
                 }
                 if (self->saved_session_type != NULL)
                         set_session_type (self, self->saved_session_type);
+                else
+                        update_session_type (self);
         }
 
 }
-- 
GitLab

